src/Controller/Api/ClinicController.php

```diff
@@ -87,30 +79,25 @@ public function edit(Request $request, $clinic)
         $clinic->setName($request->get('name'));
         $clinic->setAddress($address);
 
-        if (!empty($clinic->getName())) {
-            $entityManager = $this->getDoctrine()->getManager();
-            $entityManager->flush();
-
-            return new JsonResponse(['msg' => 'Clinic edited whit success!'], Response::HTTP_OK);
-        }
-
-        return new JsonResponse(['msg' => 'Check the empty fields'], Response::HTTP_NOT_ACCEPTABLE);
+        $form = $this->createForm(ClinicType::class, $clinic);
+        $form->submit($clinic);
+        
+        $entityManager = $this->getDoctrine()->getManager();
+        $entityManager->flush();
+        
+        $response = new JsonResponse(['msg'=>'Clinic edited whit success!'], Response::HTTP_OK);
+        
+        return $response;
     }
 
-    public function delete($clinic)
+    public function delete(Clinic $clinic)
     {
-        if (empty($clinic)) {
-            return new JsonResponse(['msg' => 'Clinic not found!'], Response::HTTP_NOT_FOUND);
-        }
-        
-        if (!empty($clinic)) {
-            $entityManager = $this->getDoctrine()->getManager();
-            $entityManager->remove($clinic);
-            $entityManager->flush();
-
-            return new JsonResponse(['msg' => 'Clinic deleted whit success!'], Response::HTTP_OK);
-        }
+        $entityManager = $this->getDoctrine()->getManager();
+        $entityManager->remove($clinic);
+        $entityManager->flush();
 
-        return new JsonResponse(['msg' => 'We could not find'], Response::HTTP_NOT_ACCEPTABLE);
+        $response = new JsonResponse(['msg'=>'Clinic deleted whit success!'], Response::HTTP_OK);
+        
+        return $response;
```

Comment:
This `return $response` is unnecessary. You can just return the new  `JsonResponse` object, like this:
```suggestion
        return new JsonResponse(['msg' => 'We could not find'], Response::HTTP_OK);
```